demo/src/main/java/com/project/demo/rest/AuthRestController.java

```diff
@@ -84,4 +88,38 @@ public ResponseEntity<?> registerUser(@RequestBody User user) {
         User savedUser = userRepository.save(user);
         return ResponseEntity.ok(savedUser);
     }
+
+    @PostMapping("/reset-password")
+    public ResponseEntity<?> resetPassword(@RequestBody User user) {
+        Optional<User> optionalUser = userRepository.findByEmail(user.getEmail());
```

Comment:
Please try to use a service layer instead.

Reply:
Iván updated his class. He is now using the service layer to then interact with the repository.